core/src/main/java/org/apache/calcite/sql/dialect/PostgresqlSqlDialect.java

```diff
@@ -180,23 +181,28 @@ public PostgresqlSqlDialect(Context context) {
 
   @Override public void unparseCall(SqlWriter writer, SqlCall call,
       int leftPrec, int rightPrec) {
-    switch (call.getKind()) {
-    case FLOOR:
-      if (call.operandCount() != 2) {
-        super.unparseCall(writer, call, leftPrec, rightPrec);
-        return;
-      }
+    if (call.getOperator() == SqlStdOperatorTable.LISTAGG) {
+      RelToSqlConverterUtil.specialOperatorByName("STRING_AGG")
```

Comment:
Maybe can implement like this:
```
SqlCall endsWithCall = SqlLibraryOperators.ENDSWITH
          .createCall(SqlParserPos.ZERO, call.getOperandList());
      super.unparseCall(writer, endsWithCall, leftPrec, rightPrec);
```

Reply:
At first I want to implement it as your suggest way:
sql
```
    final String query = "SELECT \"product_class_id\","
        + "LISTAGG(DISTINCT CAST(\"brand_name\" AS VARCHAR), ',') "
        + "FROM \"foodmart\".\"product\" group by \"product_class_id\"";
```
would convert to :
```
SELECT \"product_class_id\", STRING_AGG(CAST(\"brand_name\" AS VARCHAR), ',')\nFROM \"foodmart\".\"product\"\nGROUP BY \"product_class_id\"
```
would loss DISTINCT.(the test case is provided in JIRA: https://issues.apache.org/jira/browse/CALCITE-6866)
so I change the way like this pr.  @NobiGo